pkg/operator/controller/controller_router_deployment.go

```diff
@@ -0,0 +1,245 @@
+	env = append(env, corev1.EnvVar{Name: "ROUTER_METRICS_TLS_CERT_FILE", Value: filepath.Join(certsVolumeMountPath, "tls.crt")})
+	env = append(env, corev1.EnvVar{Name: "ROUTER_METRICS_TLS_KEY_FILE", Value: filepath.Join(certsVolumeMountPath, "tls.key")})
+
+	if len(ingressDomain) > 0 {
+		env = append(env, corev1.EnvVar{Name: "ROUTER_CANONICAL_HOSTNAME", Value: ingressDomain})
+	}
+
+	if haType == ingressv1alpha1.CloudClusterIngressHA {
+		// For now, check if we are on AWS. This can really be done for
+		// for any external [cloud] LBs that support the proxy protocol.
+		if infraConfig.Status.Platform == configv1.AWSPlatform {
+			env = append(env, corev1.EnvVar{Name: "ROUTER_USE_PROXY_PROTOCOL", Value: "true"})
+		}
+	}
+
+	if ci.Spec.NodePlacement != nil {
+		if ci.Spec.NodePlacement.NodeSelector != nil {
+			nodeSelector, err := metav1.LabelSelectorAsMap(ci.Spec.NodePlacement.NodeSelector)
+			if err != nil {
+				return nil, fmt.Errorf("clusteringress %q has invalid spec.nodePlacement.nodeSelector: %v",
+					ci.Name, err)
+			}
+
+			deployment.Spec.Template.Spec.NodeSelector = nodeSelector
+		}
+	}
+
+	if ci.Spec.NamespaceSelector != nil {
+		namespaceSelector, err := metav1.LabelSelectorAsSelector(ci.Spec.NamespaceSelector)
+		if err != nil {
+			return nil, fmt.Errorf("clusteringress %q has invalid spec.namespaceSelector: %v",
+				ci.Name, err)
+		}
+
+		env = append(env, corev1.EnvVar{
+			Name:  "NAMESPACE_LABELS",
+			Value: namespaceSelector.String(),
+		})
+	}
+
+	replicas := ci.Spec.Replicas
+	deployment.Spec.Replicas = &replicas
+
+	if ci.Spec.RouteSelector != nil {
+		routeSelector, err := metav1.LabelSelectorAsSelector(ci.Spec.RouteSelector)
+		if err != nil {
+			return nil, fmt.Errorf("clusteringress %q has invalid spec.routeSelector: %v", ci.Name, err)
+		}
+		env = append(env, corev1.EnvVar{Name: "ROUTE_LABELS", Value: routeSelector.String()})
+	}
+
+	deployment.Spec.Template.Spec.Containers[0].Env = append(deployment.Spec.Template.Spec.Containers[0].Env, env...)
+
+	deployment.Spec.Template.Spec.Containers[0].Image = routerImage
+
+	if haType == ingressv1alpha1.UserDefinedClusterIngressHA {
+		// Expose ports 80 and 443 on the host to provide endpoints for
+		// the user's HA solution.
+		deployment.Spec.Template.Spec.HostNetwork = true
+
+		// With container networking, probes default to using the pod IP
+		// address.  With host networking, probes default to using the
+		// node IP address.  Using localhost avoids potential routing
+		// problems or firewall restrictions.
+		deployment.Spec.Template.Spec.Containers[0].LivenessProbe.Handler.HTTPGet.Host = "localhost"
+		deployment.Spec.Template.Spec.Containers[0].ReadinessProbe.Handler.HTTPGet.Host = "localhost"
+	}
+
+	// Fill in the default certificate secret name.
+	secretName := fmt.Sprintf("router-certs-%s", ci.Name)
+	if ci.Spec.DefaultCertificateSecret != nil && len(*ci.Spec.DefaultCertificateSecret) > 0 {
```

Comment:
Hidden dependency on the default cert?

Reply:
Could use `routerDefaultCertificateSecretName` to get the name—would that be copacetic?

Reply:
Oh, I guess you're saying that `ensureClusterIngress` should pass the secret from `ensureDefaultCertificateForIngress` as an input to `ensureRouterDeployment`?

Reply:
That would be a circular dependency (deployment → secret → deployment), which would necessitate a bit more thought and refactoring to sort out.

Reply:
Yeah, not recommending any action with this PR, just wanted to note it stood out at me as something we missed and need to talk more about